gp/local_search/gradient_based.py

```diff
@@ -1,12 +1,28 @@
 import numpy as np
 
 try:
-    import torch
+    import torch  # noqa: F401
+
+    torch_available = True
 except ModuleNotFoundError:
-    torch = None
+    torch_available = False
+
+from typing import Callable, List, Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import torch.optim.Optimizer
+
+from ..individual import Individual  # noqa: F401
 
 
-def gradient_based(individuals, objective, lr, gradient_steps, optimizer=None, clip_value=None):
+def gradient_based(
+    individuals: List[Individual],
+    objective: Callable,
```

Comment:
```suggestion
    objective: Callable[[Individual], Individual],
```

Reply:
Should be this Callable[[torch.nn.Module], torch.Tensor]. Done.